### ISSUE TITLE
Why is KL constraint disabled (kl_loss_coef=0.0) during RL training?

### ISSUE THREAD

**Issue by 4NLLab** [TIGER-AI-Lab/Pixel-Reasoner#36]
Hi authors,

Thanks for your great work on Pixel Reasoner. I am currently trying to reproduce the paper and have a question regarding the RL training configuration.

I noticed that in the original training script, --init_kl_coef was set to 0.0.

Furthermore, in the new version under the Verl tool framework, the training configuration also sets:

kl_loss_coef=0.0

coef=0

entropy_coeff=0

This seems to disable all KL and entropy constraints, which are typically used to prevent the policy (actor) from diverging too far from the reference model (SFT model) and to encourage exploration.

In my own experiments following this setup, I observed that the actor/kl_loss (the divergence between the actor and the reference model) keeps increasing significantly during training. This makes me wonder if this is the expected behavior.

Could you please explain the reasoning behind this setting (disabling KL constraints)? How do you ensure policy stability and prevent it from collapsing or "reward hacking" without these constraints?

Thank you for your time and clarification!

**Comment by cjakfskvnad** [comment]
Thank you for the great question. 
We disable kl constraints to improve the performance because with kl constraints, model might explore less and try to stay close to the reference model. In the experiments, we found without self-correction data in sft stage, the model showed behavior of reward-hacking: The model outputs funny arguments of visual operations such as selecting negative frames to get the curiosity reward and make the video stay the same. However, with self correction data it will not show this behavior.
